### PR TITLE
Functioning api

### PR DESCRIPTION
**Pull request by aasnani**
Please go easy on me rawr XD

### CODE REVIEW

**Review by sjakati98 — COMMENTED** [review]
Nice!

File: server.js [hunk 1]
```diff
@@ -16,4 +16,178 @@ MongoClient.connect('mongodb://localhost', { useNewUrlParser: true }).then(conne
...
+      // Adding/reassigning/initializing properties of video object to be added to our collection
+      newVideo.created = new Date();
+      newVideo.upvotes = 0;
+      newVideo.downvotes = 0;
+      db.collection('videos').insertOne(newVideo).then(result => //Adding it to our collection
+        db.collection('videos').find({_id: result.insertedId}).limit(1).next() //finding it again to return it as part of the JSON
+      ).then(newVideo => {
+        res.json({
+          success: true, //if we do add it, we set the response body success property to true
+          video: {
```
Comment (seangtkelley) [inline server.js:108]: Could this just be like
```suggestion
          video: newVideo
```

File: server.js [hunk 2]
```diff
@@ -16,4 +16,178 @@ MongoClient.connect('mongodb://localhost', { useNewUrlParser: true }).then(conne
...
+  ).then(result => {
+    if(result.matchedCount === 0 || result.modifiedCount === 0) { //if no video object matches what we are trying to update, we just set success to false and return that
+      res.json({success: false});
+    }
+    else{ //if we do find it
+      db.collection('videos').find({wikiPageId: video.wikiPageId, sectionIdx: video.sectionIdx, ytId: video.ytId}).limit(1).next() //finding the updated video again
+      .then(updatedVideo => {
+        res.json({
+          success: true, //true because we could update it
+          video: {
```
Comment (seangtkelley) [inline server.js:179]: Same comment as above.

**Review by seangtkelley — CHANGES_REQUESTED** [review]
I'm not sure if my comments below are applicable but overall the good looks good.

File: server.js [hunk 3]
```diff
@@ -16,4 +16,178 @@ MongoClient.connect('mongodb://localhost', { useNewUrlParser: true }).then(conne
...
+*/
+
+app.post('/api/videos', (req,res) => {
+  let newVideo = req.body.video;
+
+  //first we search the database to see if the video object already exists
+  db.collection('videos').find({wikiPageId: newVideo.wikiPageId, sectionIdx: newVideo.sectionIdx, ytId: newVideo.ytId}).count().then(count => {
+    if(count != 0){ //if it already exists, we don't add it
+      res.json({
+        success: false //if we don't add it, we set the response body success property to false
```
Comment (aasnani) [inline server.js:95]: add error to here so we can display error to users

File: server.js [hunk 4]
```diff
@@ -16,4 +16,178 @@ MongoClient.connect('mongodb://localhost', { useNewUrlParser: true }).then(conne
...
+            sectionIdx: newVideo.sectionIdx,
+            ytId: newVideo.ytId, //constructs simplified video object to be returned to client with only necessary information
+            upvotes: newVideo.upvotes,
+            downvotes: newVideo.downvotes
+          }
+        }
+        );
+      }).catch(error => {
+        console.log(error);
+        res.status(500).json({ message: `Error in parsing POST request to /api/videos/ with video youtube id ${newVideo.ytId} and wikipage id ${newVideo.wikiPageId}: ${error}`});
```
Comment (aasnani) [inline server.js:119]: get rid of status, just add error to response object

**Pull request merged**